### PR TITLE
SpatialBinaryLogisticRegression criterion

### PR DESCRIPTION
**Pull request by sagarwaghmare69**
Added SpatialBinaryLogisticRegression criterion along with unit test.

### CODE REVIEW

File: test/test.lua [hunk 1]
```diff
@@ -1132,6 +1132,66 @@ function dpnntest.WhiteNoise()
    mytester:assertTensorEq(gradOutput, gradInput, 0.000001, "WhiteNoise backward err")
 end
 
+-- Unit Test SpatialBinaryLogisticRegression criterion
+function dpnntest.SpatialBinaryLogisticRegression()
+   print("Testing SpatialBinaryLogisticRegression criterion")
```
Comment (nicholas-leonard) [inline test/test.lua]: Can you remove this print?

**Comment by nicholas-leonard** [comment]
Unit test and code looks good. Could you add some documentation for this module in the README?

File: SpatialBinaryLogisticRegression.lua [hunk 2]
```diff
@@ -0,0 +1,84 @@
+------------------------------------------------------------------------
+--[[ SpatialBinaryLogisticRegression ]]--
+-- Takes an image of size batchSize x nChannel x width x height as input.
+-- Computes Binary Logistic Regression Cost.
+-- Useful for 2 class pixel classification.
+------------------------------------------------------------------------
+
+local SpatialBinaryLogisticRegression, parent = torch.class('nn.SpatialBinaryLogisticRegression', 'nn.Criterion')
+
+function SpatialBinaryLogisticRegression:__init(sizeAverage)
```
Comment (nicholas-leonard) [inline SpatialBinaryLogisticRegression.lua]: can you remove sizeAverage from constructor argument. Just make it `self.sizeAverage = true`. If the user wants to change that, he can do so by setting it via 

``` lua
criterion = SpatialBinaryLogisiticRegression() 
criterion.sizeAverage = false
```

This is standard practice for nn (MSECriterion, being the exception I think).

**Comment by sagarwaghmare69** [comment]
Added the criterion to README. Corrected the cost by adding division by 2 to it.

**Comment by nicholas-leonard** [comment]
Thanks @sagarwaghmare69

**Pull request merged**